packages/babel-generator/src/printer.ts

```diff
@@ -1004,7 +1012,19 @@ class Printer {
           hasLoc = false;
 
           if (len === 1) {
-            this._printComment(comment, COMMENT_SKIP_NEWLINE.SKIP_ALL);
+            if (type === COMMENT_TYPE.LEADING) {
+              this._printComment(
+                comment,
+                (!isStatement(node) && !isProperty(node)) ||
+                  isFunction(parent, { body: node })
+                  ? COMMENT_SKIP_NEWLINE.SKIP_ALL
+                  : COMMENT_SKIP_NEWLINE.DEFAULT,
+              );
+            } else if (type === COMMENT_TYPE.TRAILING && isLVal(node)) {
```

Comment:
```suggestion
            } else if (type === COMMENT_TYPE.TRAILING && (isLVal(node) || isExpression(node))) {
```

Or maybe `!isStatement(node) && !isClassBody(parent) && !isTSInternfaceBody(parent)`, since intuitively looking at our tests this makes sense as "not statements or things in classes".

Reply:
The results are all good.

To be honest, I'm not sure I'm on the right path, but there doesn't seem to be a better way. Hopefully, there will not be too many such special judgments in the future.